src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -266,6 +266,47 @@ public void WriteValue(object value)
             }
         }
 
+        public void WriteValue(ReadOnlySpan<byte> span)
+        {
+            int length = span.Length;
+            switch (length)
+            {
+                case 0:
+                    _writer.Write((byte)EncodingKind.Array_0);
+                    break;
+                case 1:
+                    _writer.Write((byte)EncodingKind.Array_1);
+                    break;
+                case 2:
+                    _writer.Write((byte)EncodingKind.Array_2);
+                    break;
+                case 3:
+                    _writer.Write((byte)EncodingKind.Array_3);
+                    break;
+                default:
+                    _writer.Write((byte)EncodingKind.Array);
+                    WriteCompressedUInt((uint)length);
+                    break;
+            }
+
+            var elementType = typeof(byte);
+            Debug.Assert(s_typeMap[elementType] == EncodingKind.UInt8);
+
+            WritePrimitiveType(elementType, EncodingKind.UInt8);
+
+#if NETCOREAPP
+            _writer.Write(span);
+#else
+            var buffer = new byte[Math.Min(length, 8192)];
```

Comment:
Why 8192?

Reply:
Could this be ArrayPool instead of `new byte`?

Reply:
> Why 8192?

A power of 2 large enough to hopefully be efficient for block memory transfers but small enough to not be in the LOH. It was somewhat arbitrarily chosen since it's a legacy code path.

> Could this be ArrayPool instead of new byte?

Possibly, but I'm not sure it matters. Previously we were generating a larger buffer, but now we are generating a smaller one.

Reply:
> It was somewhat arbitrarily chosen since it's a legacy code path.

Please document this fact.

Reply:
> Please document this fact.

Will do as a follow-up PR unless some other change requires this PR to be rebuilt and miss the snap.

Reply:
Now documented.